src/common/signal_handling.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+#include <csignal>
+#include <string>
+
+// SIGNAL HANDLING
+
+// The signal handlers (and checkers) here are implemented in line with with the recommendations
+// for signal handling in the SEI CERT C Coding Standard, specifically
+//
+// - SIG30-C:
+//   https://wiki.sei.cmu.edu/confluence/display/c/SIG30-C.+Call+only+asynchronous-safe+functions+within+signal+handlers
+//
+// - SIG31-C:
```

Comment:
Does this mean that sig=31 is a valid value? If so, please use unsigned types in the bit masks, e.g. 1U << sig

Reply:
1. No. SIG31-C is just the number of a rule 
2. Signals go up to 64, but 32 and above (with 32 and 33 actually reserved for thread management) are part of the POSIX real time extensions and not of interest to us. Neither is 30 (system lost power) or 31 (wrong argument provided to system call). Since sig_atomic_t  is a signed it, I've limited the range to 1-30. No. 29 (SIGIO) might be of interest at some point (e.g. in a server that uses sockets).